### PR TITLE
Test profile: add nf-tower for nf-core@mailinator.com

### PR DESCRIPTION
**Pull request by ewels**
Added nf-tower credentials for nf-core@mailinator.com to the test profile config file.

Should allow anyone to log in to nf-tower using the [public mailbox](https://www.mailinator.com/v3/index.jsp?zone=public&query=nf-core) and see the status of Travis jobs. ✨

### CODE REVIEW

**Comment by drpatelh** [comment]
Think we may have to disable `continuous-integration/travis-ci` here in the branch rules because it most likely won't ever run!

**Comment by drpatelh** [comment]
Sorry! Premature approval :sweat_smile: I'll wait for a review request next time!

**Comment by ewels** [comment]
> I'll wait for a review request next time!

You did nothing wrong! Should have labelled as `WIP` sorry..

**Comment by ewels** [comment]
NB: Tests will fail on this now as I've messed around with the travis file enough that it fails in the linting

**Comment by ewels** [comment]
@drpatelh / @apeltzer / @jfy133 - how does this compare to your `eager` implementation?

I'm fairly happy with this now.. Generates job names like:

* `methylseq-pull_request-104-b7ae78-bismark-skiptrim-rrbs`
* `methylseq-pull_request-104-b7ae78-bismark-withref`
* `methylseq-push-false-c6cbc9-bwameth-skip`
  * This is when a push and not a PR - could probably do some bash variable magic to remove the `-false`

Other things I considered:

* Auth and usage is only in the travis config and not in the test profile config, so people running the test profile themselves should be unaffected (eg. offline clusters)
* Cleaned up Travis environment variables into `global` and `job` so that the test commands are not so ridiculously long
    * Note that the linter would need to be updated to support this (hence tests failing at the moment).

How does this compare to what you guys did? How should we go forward?

Phil

**Comment by jfy133** [comment]
I didn't get as far as you, but I was going along the same lines (my next step was to move from test configs to the travis env, like you've done and then continue testing).

But I agree `<pipline_name>-<vaious_travis_info>-<short_test_description>` is intuitive. 

To keep things shorter I didn't have the commit, but rather the travis build ID as it makes it easier to find in the travis website, I find - which is usually what I'm having to debug. It would make the name shorter too.

I'll make a proper test PR onto EAGER and continue testing (I was running forward a bit and using it in a separate PR where it was useful for bugging), so can give a bit more info then.

**Comment by ewels** [comment]
> To keep things shorter I didn't have the commit, but rather the travis build ID as it makes it easier to find in the travis website, I find - which is usually what I'm having to debug. It would make the name shorter too.

Ah ok, nice info to have 👍  I quite like having the commit hash though (in addition), as this is what shows in the GitHub PR interface so means that you can skip going to Travis if you think it's a run that has failed.

**Comment by apeltzer** [comment]
Didn't have time yet to have a look but this looks quite amazing already! James please go ahead in eager, I'll review whenever I have time 😓

**Comment by ewels** [comment]
Tests passing and commit history clean - good to merge now I think @drpatelh!

**Comment by drpatelh** [comment]
Question. I think you may have tried this already but is there a way to add the following variables as global instead of repeating them on each line?

```
TOWER_ACCESS_TOKEN="1c1f493bc2703472d6f1b9f6fb9e9d117abab7b1"
JOBNAME="methylseq-${TRAVIS_EVENT_TYPE}-${TRAVIS_PULL_REQUEST}-${TRAVIS_COMMIT:0:6}-"
NXF_ANSI_LOG=0
```

If not, then feel free to merge 👍

**Comment by jfy133** [comment]
I'm trying this, sort of along those lines

**Comment by ewels** [comment]
Yeah, I did exactly that and tidied up the consistent variables under the Travis `global` scope. See the code in b4804b5693dc9f80a89f49c0195388ca6ba00de1:

https://github.com/nf-core/methylseq/blob/b4804b5693dc9f80a89f49c0195388ca6ba00de1/.travis.yml#L32-L64

This worked great, but the nf-core linting failed because the structure of the Travis `env` dictionary is different, so it thought that the code wasn't testing the older version of Nextflow.

I figured that we're dropping Travis in favour of GitHub Actions soon anyway, so I couldn't really be bothered to change the linting and wait for a tools release. So I just reverted the tidying to get all of the tests to pass again.

File: .travis.yml [hunk 1]
```diff
@@ -28,23 +28,23 @@ install:
...
-  
+
 env:
-  - ALIGNER=bismark ALIGNER_REF="--bismark_index ${TRAVIS_BUILD_DIR}/tests/results/reference_genome/BismarkIndex/" NXF_VER='19.04.0'
-  - ALIGNER=bismark ALIGNER_REF="--bismark_index ${TRAVIS_BUILD_DIR}/tests/results/reference_genome/BismarkIndex/"
-  - ALIGNER=bismark_hisat ALIGNER_REF="--bismark_index ${TRAVIS_BUILD_DIR}/tests/results/reference_genome/BismarkIndex/" NXF_VER='19.04.0'
-  - ALIGNER=bismark_hisat ALIGNER_REF="--bismark_index ${TRAVIS_BUILD_DIR}/tests/results/reference_genome/BismarkIndex/"
-  - ALIGNER=bwameth ALIGNER_REF="--bwa_meth_index ${TRAVIS_BUILD_DIR}/tests/results/reference_genome/genome.fa" NXF_VER='19.04.0'
-  - ALIGNER=bwameth ALIGNER_REF="--bwa_meth_index ${TRAVIS_BUILD_DIR}/tests/results/reference_genome/genome.fa"
+  - TOWER_ACCESS_TOKEN="1c1f493bc2703472d6f1b9f6fb9e9d117abab7b1" JOBNAME="methylseq-${TRAVIS_EVENT_TYPE}-${TRAVIS_PULL_REQUEST}-${TRAVIS_COMMIT:0:6}-" NXF_ANSI_LOG=0 ALN=bismark       AREF="--bismark_index ${TRAVIS_BUILD_DIR}/tests/results/reference_genome/BismarkIndex/" NXF_VER='19.10.0'
```
Comment (jfy133) [inline .travis.yml]: Small warning here: by putting the access token here, all forks will also inherit this. Therefore, every user running their own travis runs will also end up in the nf-core tower.

This is solved with transfer to github actions, as you can use the secret token thing.

Comment (ewels) [reply .travis.yml]: Yes, good point. Travis can also do secrets. I think instead we can set the flag in an env variable conditionally based on the root user / organisation though.

**Comment by ewels** [comment]
This seems like a long time ago, editing config files for Travis!

Anyway, PR can be closed now. Should hopefully have pretty first-class support in Tower very soon.

**Pull request closed without merging**